Adjust dashboard to be mobile friendly

hulloanson opened the pull request:
Key changes:

Main breakpoint is when screen size is smaller than a tablet (i.e. less than 1024px). At this smaller size:
- Sidebar is placed at the bottom
- Popups do not show as long as the screen height is 500px or greater
- Header links are collapsed into a hamburger menu

Also, in all views, the selected map pin is now displayed at a larger size than the other surrounding pins to help with differentiation

hulloanson committed:
added desktop and mobile styles; removed redundant ampersands

hulloanson committed:
put sidebar at the bottom when screen width is smaller than a tablet

hulloanson committed:
fixed scrolls in "sidebar" a.k.a. bottom bar in mobile view

hulloanson committed:
eliminated extraneous scrollbars

hulloanson committed:
collapse header; include tablet in mobile version

hulloanson committed:
adjusted map / panel size for different screen sizes

hulloanson committed:
have leaflet recalculate map size upon container size change

hulloanson committed:
fixed feed item not entirely visible

tallcoleman commented:
This is already looking really cool @hulloanson !

hulloanson committed:
misc refactor and adjusted spaces of different parts of the page

hulloanson committed:
display map popup again when the screen is very short

tallcoleman committed:
CSS cleanup

- map folder reduced back to one file
- removed duplicate code
- fixed import paths

tallcoleman commented:
I'm not sure if we should remove the `&` symbol yet for nested CSS - e.g. for Chrome, the browser versions from between April and October 2023 still require it: https://caniuse.com/css-nesting

tallcoleman commented on bikespace_dashboard/css/components/page/mobile.css:
```diff
@@ -0,0 +1,114 @@
+:root {
+  /* preferred visible area of map.
+   * taller screens takes advantage of this when the panel is open
+   * to show a bigger map
```
Does this make the map bigger or the panel? It looks like it increases the panel size when more space is available.

tallcoleman reviewed (commented):
## Notes from review:

### Must fix before merge

- [x] Nav bar disappears on desktop view when a pin is focused
- [x] Can't click on map to un-focus a pin
- [x] Since the feed is the only data shown in mobile view, it should show all the relevant info for a report (need to add parking duration and issue id)

### Nice to have 

- [ ] Change drawer button to an up/down chevron to make interaction intent clearer
- [ ] Make map credits smaller in mobile view (e.g. [maplibre uses a toggle](https://maplibre.org/maplibre-gl-js/docs/examples/add-a-marker/))
- [ ] Sidebar content is very wide just below the mobile breakpoint - should consider either having a multi-column flex layout or making the breakpoint smaller

tallcoleman committed:
More CSS cleanup

- Thought that `width < 1024px` would communicate the intent of having a 1024px breakpoint more clearly
- Fixed some styling that was interfering with the intended styling for the Leaflet map

tallcoleman committed:
Clicking on map unsets big marker

Might still need to think through whether the issue list should change on popup close

tallcoleman committed:
Clicking map now unfocuses submission in sidebar

tallcoleman committed:
Interaction improvements for issue focus/unfocus

- enforced private methods and fields with # prefix and moved public methods top of classes
- removed focusSubmission method in SharedState - focusing interaction fully handled by hash router instead
- popup anchor adjusts to icon size
- clicking marker only automatically focuses in sidebar on mobile

tallcoleman committed:
Centralized app state management through hash router

Removed last instance of components managing each others' state directly

tallcoleman committed:
Update README.md

tallcoleman commented:
@hulloanson "Can't click on map to un-focus a pin" from the review should now be resolved. Also made a number of edits to try to centralize all management of app state through the hash router.

tallcoleman committed:
Added full details to sidebar feed items

- Added (desired) parking duration and issue id
- Sorted issue chips by render priority
- Updated styling

tallcoleman commented:
@hulloanson fixed this one, now just "Nav bar disappears on desktop view when a pin is focused" which needs fixing before merge

> Since the feed is the only data shown in mobile view, it should show all the relevant info for a report (need to add parking duration and issue id)

hulloanson committed:
fixed nav bar disappearing

hulloanson commented:
> @hulloanson fixed this one, now just "Nav bar disappears on desktop view when a pin is focused" which needs fixing before merge

fixed. check it out :)

tallcoleman committed:
Fix font size on issue tags

tallcoleman committed:
Simplified unfocus submission on map click

tallcoleman committed:
Bugfix - scrollIntoView hid feed title

When a sidebar feed item was selected, scrollIntoView is called but this was hiding the feed title bar because programmatic scrolling can move nodes out of the viewable area if overflow-y is set to hidden. Set overflow-y to clip instead on the relevant elements.

tallcoleman committed:
Amending scrollIntoView bugfix

tallcoleman reviewed (approved):
@hulloanson ready to merge I think! I just added some final tweaks. You were right about how challenging it is to get `scrollIntoView()` working properly - have added some changes to prevent it from hiding the feed title, but if you think there's a better way of fixing that problem, let me know.
